rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -81,17 +83,30 @@ template<typename MessageT, typename Alloc, typename PublisherT>
 std::shared_ptr<PublisherT>
 Node::create_publisher(
   const std::string & topic_name, const rmw_qos_profile_t & qos_profile,
-  std::shared_ptr<Alloc> allocator)
+  std::shared_ptr<Alloc> allocator, IntraprocessSetting use_intra_process_comm)
 {
   if (!allocator) {
     allocator = std::make_shared<Alloc>();
   }
 
+  bool use_intra_process;
+  switch (use_intra_process_comm) {
+    case IntraprocessSetting::ENABLE:
+      use_intra_process = true;
+      break;
+    case IntraprocessSetting::DISABLE:
+      use_intra_process = false;
+      break;
+    default:  // IntraprocessSetting::DEFAULT_FROM_NODE
```

Comment:
This should be its own case instead, and an error should be thrown if yet something else is used.